src/Chapter2.hs

```diff
@@ -858,7 +885,15 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind [] = []
+rewind [x] = [x]
+rewind (x:xs) = go [x] xs
```

Comment:
You don't need special cases here 🙂 I believe, if you just define `rewind` in terms of eta-reduce helper `go` function, it will work just fine:
```suggestion
rewind = go []
```

Reply:
You right, it works, I think this is amazing, it's just a bit hard to think this way